pandas/core/arrays/datetimes.py

```diff
@@ -209,6 +204,15 @@ def __new__(cls, values, freq=None, tz=None, dtype=None):
         # if dtype has an embedded tz, capture it
         tz = dtl.validate_tz_from_dtype(dtype, tz)
 
+        if isinstance(values, DatetimeArrayMixin):
+            values = values.asi8
```

Comment:
why getting the integers here and not M8 directly?

Reply:
Because the way the constructors view i8 values is more consistent.  For tz-naive, M8 vs i8 are equivalent.  For tz-aware, i8 is interpreted as unix timestamps (i.e. UTC), whereas M8 are interpreted as the wall-time in the given timezone.